Configure the project to run with Docker

kleberngueriero opened the pull request:
Add Dockerfile and docker-compose.yml to run the project and makes the configuration changes to run the project in Docker.
Also, add a docker-compose.override.yml that makes it easier to use in development.

Relative to issue https://github.com/call4paperz/call4paperz/issues/185

lucasrenan commented on Dockerfile:
```diff
@@ -0,0 +1,12 @@
+FROM ruby:2.3.1
```
dumb question, which image is this one? which os?

kleberngueriero replied:
I knew it was a debian or debian based, but just checked: It's Debian GNU/Linux 8(jessie).

lucasrenan commented:
@kleberngueriero please add the issue in the PR description: https://github.com/call4paperz/call4paperz/issues/185

lucianosousa commented:
What about use `letter_opener` instead of `mailcatcher` ?
What do u guys think?

Less config, always better.

lucasrenan commented:
@lucianosousa I'd say `letter_opener` will not work out with docker, as docker containers don't know how to launch a browser, which is is the host machine.

kleberngueriero commented:
I believe would be ok to use letter_openner_web

lucasrenan commented:
@kleberngueriero yeah, it may work with `letter_opener_web`. give it a try :)

lucianosousa commented:
@jhonathas can you plz review this PR?
you PR was rejected with this idea in the past but I think we can't stop the `future` here.

I see with good eyes nowadays

kleberngueriero commented:
@lucascaton and @lucianosousa, I commited the replacement of mailcatcher with letter_opener_web.
Also, since I was making changes:

- I renamed the docker-compose.override.yml to docker-compose.override.yml.sample, so anyone can edit it for personal use.
- I changed Dockerfile to make it easier to sync bundle installation in the host(avoiding redownload of all gems everytime one need to bundle install)

lucascaton commented:
@kleberngueriero wrong Lucas 😄

cc: @lucasrenan

kleberngueriero commented:
Oh man..sorry about that :sweat_smile:

lucasrenan commented:
@lucascaton sorry man, it's always happening 😂

lucascaton commented:
@lucasrenan no problem! I've contributed to this project before, it might be the reason why GitHub keeps suggesting my username in mentions 😬

lucasrenan commented:
@jhonathas @lucianosousa any concerns regarding this PR? may we merge it?

tauil commented:
I'm not a 100% aware about it, but as far as I know about Docker, it's fine to me. 👍

lucianosousa commented:
@kleberngueriero merged. thanks!

kleberngueriero commented:
@lucianosousa, should I delete the branch?

lucasrenan commented:
@kleberngueriero go for it!

lucianosousa commented:
@kleberngueriero I think I already did it here
